src/mqt/bench/benchmark_generation.py

```diff
@@ -443,50 +452,62 @@ def get_benchmark(
-        raise ValueError(msg)
+        qc_algo = lib.create_circuit(circuit_size)
 
-    independent_level = 1
-    if level in ("indep", independent_level):
-        assert compiler_settings.qiskit is not None
-        opt_level = compiler_settings.qiskit.optimization_level
-        return get_indep_level(qc, circuit_size, opt_level, False, True)
-
-    native_gates_level = 2
-    if level in ("nativegates", native_gates_level):
-        assert compiler_settings.qiskit is not None
-        opt_level = compiler_settings.qiskit.optimization_level
-        return get_native_gates_level(qc, target, circuit_size, opt_level, False, True)
+    # Process the circuit to the specified level
+    processed_qc: QuantumCircuit
 
-    mapped_level = 3
-    if level in ("mapped", mapped_level):
-        assert compiler_settings.qiskit is not None
+    if level in ("alg", 0):
+        processed_qc = qc_algo
+    else:
+        if compiler_settings is None:
+            compiler_settings = CompilerSettings(QiskitSettings())
+        elif not isinstance(compiler_settings, CompilerSettings):
+            msg = "compiler_settings must be of type CompilerSettings or None."  # type: ignore[unreachable]
+            raise ValueError(msg)
+        assert compiler_settings.qiskit is not None, "Qiskit compiler settings are required."
         opt_level = compiler_settings.qiskit.optimization_level
-        assert isinstance(opt_level, int)
-        return get_mapped_level(
-            qc,
-            circuit_size,
-            target,
-            opt_level,
-            False,
-            True,
-        )
 
-    msg = f"Invalid level specified. Must be in {get_supported_levels()}."
-    raise ValueError(msg)
+        if level in ("indep", 1):
+            processed_qc = get_indep_level(qc_algo, circuit_size, opt_level, False, True)
+        elif level in ("nativegates", 2):
+            if target is None:
+                msg = "Target must be provided for 'nativegates' level."
+                raise ValueError(msg)
+            processed_qc = get_native_gates_level(qc_algo, target, circuit_size, opt_level, False, True)
+        elif level in ("mapped", 3):
+            if target is None:
+                msg = "Target must be provided for 'mapped' level."
+                raise ValueError(msg)
+            processed_qc = get_mapped_level(qc_algo, circuit_size, target, opt_level, False, True)
+        else:  # Should not be reached
+            msg = f"Internal error: Unhandled level '{level}'."
+            raise RuntimeError(msg)  # pragma: no cover
+
+    if generate_mirror_circuit:
+        # Get the unitary part of the processed circuit (remove measurements)
+        qc_unitary_base = processed_qc.remove_final_measurements(inplace=False)
+
+        # Compute its inverse
+        qc_inverse = qc_unitary_base.inverse()
+
+        # Compose them. The circuit qc_unitary_base is modified by compose.
```

Comment:
I am fairly sure we want to insert a global barrier after the first circuit before concatenation do delimit the circuits and to try to prevent optimizers from optimizing away the mirror circuit.